### PR TITLE
Course run key getting blanked issue fixed.

### PR DESCRIPTION
**Pull request by awais786**
The Course run key is getting blanked when a course-run is being edited and saved.

ECOM-7411

### CODE REVIEW

File: course_discovery/apps/publisher/forms.py [hunk 1]
```diff
@@ -271,6 +271,12 @@ def clean_lms_course_id(self):
 
         return None
 
+    def __init__(self, *args, **kwargs):
+        is_project_coordinator_user = kwargs.pop('is_project_coordinator_user', None)
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/forms.py]: Instead you can simply pass user and check ` if not is_project_coordinator(user)`

File: course_discovery/apps/publisher/tests/test_views.py [hunk 2]
```diff
@@ -2294,7 +2294,36 @@ def test_studio_instance_with_course_team(self):
         self.new_course_run.save()
         response = self.client.get(self.edit_page_url)
 
+        self.assertContains(
+            response, '<input id="id_lms_course_id" name="lms_course_id" type="hidden"'
+        )
+        self.assertContains(response, '<a class="studio-link"')
+
+    def test_studio_instance_with_partner_cordinator(self):
```
Comment (waheedahmed) [inline course_discovery/apps/publisher/tests/test_views.py]: `project_coordinator`

File: course_discovery/apps/publisher/forms.py (hunk 1 above)
Comment (awais786) [reply course_discovery/apps/publisher/forms.py]: Yes but we have to send the is_project_coordinator_user to template also to show the studio url.

File: course_discovery/apps/publisher/tests/test_views.py (hunk 2 above)
Comment (awais786) [reply course_discovery/apps/publisher/tests/test_views.py]: :)

File: course_discovery/apps/publisher/forms.py (hunk 1 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/forms.py]: Yeah I know, but I'm talking about when you are initializing the form.

Comment (awais786) [reply course_discovery/apps/publisher/forms.py]: But in that case we have to hit the 2 queries One for intialize and other for sending to template.

Comment (waheedahmed) [reply course_discovery/apps/publisher/forms.py]: Got it!

**Comment by awais786** [comment]
@bderusha Please review.

**Review by waheedahmed — APPROVED** [review]

**Comment by awais786** [comment]
@bderusha Please review.

**Review by tasawernawaz — APPROVED** [review]

**Pull request merged**